core/src/wsv.rs

```diff
@@ -168,36 +139,72 @@ impl<W: WorldTrait> WorldStateView<W> {
         tokens
     }
 
-    /// Apply `CommittedBlock` with changes in form of **Iroha Special Instructions** to `self`.
+    /// Add new `Asset` entity.
     ///
     /// # Errors
-    /// Can fail if execution of instruction fails(should be fine after validation)
+    /// Fails if there is no account for asset
+    pub fn add_asset(&self, asset: Asset) -> Result<(), Error> {
+        let id = asset.id.account_id.clone();
+        self.modify_account(&id, move |account| {
+            account.assets.insert(asset.id.clone(), asset);
+            Ok(())
+        })
+    }
+
+    /// Add new `Domain` entity.
+    pub fn add_domain(&mut self, domain: Domain) {
+        self.world.domains.insert(domain.id.clone(), domain);
+    }
+
+    fn process_executable(&self, executable: &Executable, authority: &AccountId) -> Result<()> {
+        match executable {
+            Executable::Instructions(instructions) => {
+                instructions.iter().cloned().try_for_each(|instruction| {
+                    // Why does `execute` need to take ownership?
+                    // If it needs to clone, that could be done
+                    // internally.
+                    let events = instruction.execute(authority.clone(), self)?;
+
+                    self.produce_events(events.into_iter().map(Event::from).collect());
+                    Ok::<_, eyre::Report>(())
+                })?;
+            }
+            Executable::Wasm(bytes) => {
+                let mut wasm_runtime = wasm::Runtime::new()?;
+                wasm_runtime.execute(self, authority, bytes)?;
+            }
+        }
+        Ok(())
+    }
 
-    /// Apply [`CommittedBlock`] with changes in form of **Iroha Special Instructions** to `self`.
+    /// Apply `CommittedBlock` with changes in form of **Iroha Special
+    /// Instructions** to `self`. `trigger_recommendations` are
+    /// applied first.
```

Comment:
> Triggers might just influence other transactions in a way that they are executed before them, but do not directly do anything to other transactions.

This doesn't say when they should execute, so I decided it's better to run fist, so that execution is more balanced: 
`TITITITIT`
vs 
`IITITITITIT`